whitematteranalysis/similarity.py

```diff
@@ -117,6 +117,42 @@ def _fiber_distance_internal_use(fiber, fiber_array, threshold=0, distance_metho
         #print "similarity range :", numpy.min(distance), numpy.max(distance)        
         distance = numpy.prod(distance, 1)
         #print "overall similarity range:", numpy.min(distance), numpy.max(distance)
+    elif distance_method == 'Mean_shape':
+        
+        # sum along fiber
+        d = numpy.sum(numpy.sqrt(distance), 1)
+        # Remove effect of number of points along fiber (mean)
+        npts = float(fiber_array.points_per_fiber)
+        d = d / npts
+        # for consistency with other methods we need to square this value
+        d = numpy.square(d)
+
+        distance = numpy.sqrt(distance)
```

Comment:
can you just do the sort for the particular endpoints? Then you don't have to compute the whole distance again below. This is slow I think